lib/OpenQA/Setup.pm

```diff
@@ -328,6 +328,60 @@ sub load_plugins {
     OpenQA::Setup::update_config($server->config, @{$server->plugins->namespaces}, "OpenQA::WebAPI::Auth");
 }
 
+sub load_ui_plugins {
+    my ($server, $plugin_route) = @_;
+
+    return if (!defined $server->config->{ini_config} || !defined $server->config->{ini_config}->{v});
+
+    push @{$server->plugins->namespaces}, 'WebAPIPlugin';
```

Comment:
The namespace should certainly be prefixed (`OpenQA::WebAPIPlugin`). Or better, make it something like `OpenQA::WebAPI::Extensions`.

Reply:
My idea here is that plugins are part of external projects, thus they may be deployed independently and thus do not belong to OpenQA:: namespace. Plus I don't want to put extra requirements to folder structure, because (maybe I am wrong?) `OpenQA::WebAPI::Extensions` will force everyone to have `myproject/OpenQA/WebAPI/Extensions/MyPlugin.pm` where `OpenQA/WebAPI` are otherwise empty. Which is not a big deal, just I am not sure that such limitation really brings any real benefit.

Reply:
But they do belong to OpenQA in the greater scheme of things. And all Perl modules should be named with CPAN namespaces in mind. You're reusing `@INC` and `use/require` after all.